File: quiz/programers/level1/체육복.js
function solution(n, lost, reserve) {
  let countStudent = n - lost.length;
  lost.forEach((lostNumber, index) => {
    for (let i = 0; i < reserve.length; i++) {
      if (lostNumber === reserve[i]) {
        reserve.splice(i, 1);
        lost.splice(index, 1);
        countStudent++;
        break;
      }
    }
  });
  lost.forEach(lostNumber => {
    for (let i = 0; i < reserve.length; i++) {
      if (reserve[i] - 1 <= lostNumber && reserve[i] + 1 >= lostNumber) {
        reserve.splice(i, 1);
        countStudent++;
        break;
      }
    }
  });
  return countStudent;
}

console.log(solution(5, [1, 2], [2, 3]));

File: quiz/programers/python/level1/체육복.py
def solution(n, lost, reserve):
    loster = set(lost) - set(reserve)
    reserver = list(set(reserve) - set(lost))
    result = 0

    for i in loster:
        if i + 1 in reserver:
            reserver.remove(i + 1)
        elif i - 1 in reserver:
            reserver.remove(i - 1)
        else:
            result += 1

    return n - result


print(solution(5, [2, 4], [3]))
